extensions/src/common/extension/mixins/configurable/addCostumes/index.ts

```diff
@@ -53,6 +53,59 @@ export default function <T extends MinimalExtensionConstructor>(Ctor: T) {
       if (action === "add and set") target.setCostume(length);
     }
 
+    /**
+     * Add a costume to the current sprite based on a bitmpa input
+     * @param {Target} target (e.g. `util.target`)
+     * @param {string} bitmapImage What image to use to create the costume
+     * @param {"add only" | "add and set"} action What action should be applied
+     * - **_add only_**: generates the costume and append it it to the sprite's costume library
+     * - **_add and set_**: Both generate the costume (adding it to the sprite's costume library) and set it as the sprite's current costume
+     * @param {string?} name optional name to attach to the costume
+     */
+    async addCostumeBitmap(target: Target, bitmapImage: string, action: "add only" | "add and set", name?: string) {
+      if (!isRenderedTarget(target)) return console.warn("Costume could not be added as the supplied target wasn't a rendered target");
```

Comment:
NOTE: Can move this check and log to an internal method (@pmalacho-mit can do this)